trulens_eval/trulens_eval/utils/imports.py

```diff
@@ -682,3 +684,104 @@ def __exit__(self, exc_type, exc_value, exc_tb):
             raise ImportError(self.messages.import_error) from exc_value
 
         # Exception will be propagated unless we return True so we don't return it.
+
+
+def get_modules(path: Path, startswith: Optional[str] = None) -> Iterable[str]:
+    """Get all module names in the given path that start with the given
+    prefix.
+    
+    Args:
+        path: The path to search for modules.
+
+        startswith: If given, only modules that with this prefix are returned.
+
+    Returns:
+        Iterable of module names. These are fully qualified.
+    """
+
+    for modinfo in pkgutil.iter_modules([str(path)]):
+
+        if startswith is not None and not modinfo.name.startswith(startswith):
+            continue
+
+        yield modinfo.name
+
+        if modinfo.ispkg:
+            for submod in get_modules(path / modinfo.name, startswith=None):
+                submodqualname = modinfo.name + "." + submod
+
+                if startswith is not None and not submodqualname.startswith(
+                        startswith):
+                    continue
+
+                yield modinfo.name + "." + submod
+
+
+def get_defined_members(modname: str) -> Iterable[Tuple[str, Any]]:
+    """Get all members defined in the given module, i.e. members that are not
+    aliases with definitions somewhere else.
+    
+    A limitation is that basic types like int, str, etc. are not produced as
+    their location of definition cannot be easily checked.
+
+    Args:
+        modname: The module name.
+
+    Returns:
+        Iterable of tuples (qualname, val) where qualname is the fully qualified
+        name of the member and val is the value of the member.
+    """
+
+    try:
+        mod = importlib.import_module(modname)
+    except BaseException:
+        return
+
+    for item, val in inspect.getmembers_static(mod):
+        if item.startswith("_"):
+            continue
+
+        defmod = inspect.getmodule(val)
+
+        # Check for aliases: classes/modules defined somewhere outside of
+        # mod. Note that this cannot check for aliasing of basic python
+        # values which are not references.
+        if defmod is None or defmod.__name__ != modname:
+            continue
+
+        if len(item) == 1:
+            # Type var names.
+            continue
+
+        qualname = modname + "." + item
+        yield (qualname, val)
+
+
+def get_definitions(
```

Comment:
you don't actually use this function. What's it for?

Reply:
Moving this and related to the tests folder as it is only being used for tests (working on the test).